drivers/include/periph/freqm.h

```diff
@@ -0,0 +1,118 @@
+ *
+ * This interface allows to configure and use the for Frequency Meter (FREQM)
+ * available on some Microchip Boards.
+ *
+ * The Frequency Meter uses the frequency of a known reference clock to
+ * determine with a 24-bit accuracy the frequency of another clock.
+ *
+ *
+ * @{
+ *
+ * @file
+ * @brief       FREQM peripheral driver interface definitions
+ *
+ * @author      Urs Gompper <urs.gompper@ml-pa.com>
+ */
+
+#ifndef PERIPH_FREQM_H
+#define PERIPH_FREQM_H
+
+#include <mutex.h>
+
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Channel Enable Mask */
+#define GCLK_PCHCTRL_CHEN_Msk   (_U_(0x1) << GCLK_PCHCTRL_CHEN_Pos)
+/* Enable Mask */
+#define FREQM_CTRLA_ENABLE_Msk  (_U_(0x1) << FREQM_CTRLA_ENABLE_Pos)
+/* Start Measurement Mask */
+#define FREQM_CTRLB_START_Msk   (_U_(0x1) << FREQM_CTRLB_START_Pos)
+/* Measurement Done Interrupt Enable Mask */
+#define FREQM_INTENSET_DONE_Msk (_U_(0x1) << FREQM_INTENSET_DONE_Pos)
+/* Measurement Done Mask */
+#define FREQM_INTFLAG_DONE_Msk  (_U_(0x1) << FREQM_INTFLAG_DONE_Pos)
+/* FREQM Status Mask */
+#define FREQM_STATUS_BUSY_Msk   (_U_(0x1) << FREQM_STATUS_BUSY_Pos)
+/* Sticky Count Value Overflow Mask */
+#define FREQM_STATUS_OVF_Msk    (_U_(0x1) << FREQM_STATUS_OVF_Pos)
+
+
+typedef void (*freqm_callback_t)(void *context);
+
+/**
+ * @brief   Frequency meter configuration
+ *
+ */
+struct freqm_config {
+    uint8_t msr_clk_src;    /*< GCLK source select for measurement */
+    uint8_t ref_clk_src;    /*< GCLK source select for reference */
+    uint8_t ref_clk_cnt;    /*< Measurement duration in number of reference clock cycles */
+    uint32_t ref_clk_freq;  /*< Frequency of reference clock */
+};
+
+enum {
+    FREQM_NON_BLOCKING = 0,         /**< non-blocking mode */
+    FREQM_BLOCKING,                 /**< blocking mode */
+};
+
+/**
+ * @brief   Start frequency measurement
+ *
+ * @param[in] blocking      block if 1, don't block if 0
+ *
+ * @return                  true if measurement started successfully
+ * @return                  false if frequency measurement is already ongoing
+ */
+bool freqm_start(int blocking);
+
+/**
+ * @brief   Read number of periods of measured clock and calculate frequency
+ *
+ * @param[in] config        configuration
+ * @param[out] result       calcucalted frequency
+ *
+ * @return                  1 on error
+ * @return                  0 on success
+ */
+int freqm_frequency_get(struct freqm_config *const config, uint32_t *result);
+
+/**
+ * @brief Register a callback function
+ *
+ * @param freqmCallback     callback function
+ * @param context           pointer to user defined context data
+ */
+void freqm_callback_register(freqm_callback_t freqmCallback, void *context);
+
+/**
+ * @brief   Initialize the frequency meter
+ *
+ * @param[in] config        configuration
+ *
+ * @return                  0 on success
+ * @return                  -1 on invalid ADC line
+ */
+int freqm_init(struct freqm_config *const config);
```

Comment:
I find it a bit disturbing the place the const word between the struct and the variable name.
But I guess this is not against our coding convention ?

Reply:
I have no strong feelings on this and agree that 
```c
int freqm_init(const struct freqm_config *config);
```

looks nicer.